website/docs/r/app_service_plan.html.markdown

```diff
@@ -117,7 +117,7 @@ The following arguments are supported:
 
 ~> **NOTE:** Attaching to an App Service Environment requires the App Service Plan use a `Premium` SKU (when using an ASEv1) and the `Isolated` SKU (for an ASEv2).
 
-* `reserved` - (Optional) Is this App Service Plan `Reserved`. Defaults to `false`.
+* `reserved` - (Optional) Specifies whether it's Linux Platform. Should be `true` If Linux app service plan, false otherwise. Defaults to `false`.
```

Comment:
this isn't just for Linux - this specifies if the App Service Plan should be always running or start/stopped on demand

Reply:
This field name is very misleading. According to https://docs.microsoft.com/en-us/rest/api/appservice/appserviceplans/createorupdate#request-body, it says `If Linux app service plan true, false otherwise.`. 

According to the implementation of azure.cli: https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/appservice/custom.py#L1648
users could specify the flag "--is-linux", and azure.cli will convert it to `reserved` field.

I have also emailed the service team, they says "reserved" means the OS type, "kind" is just a ARM level field, it's only used to drive the UX. It's not needed when creating. Azure.cli doesn't have the argument "kind".

Currently I think it's a little messy for resource "azurerm_app_service_plan" to have both field "reserved" and "kind", though it's the fault of service api definition. There seems a issue fired by a member of the service team: https://github.com/terraform-providers/terraform-provider-azurerm/issues/5584

what's more, the OS type of app_service and function_app is defined by the app service plan, but for now both resource have "os_type" field, it should be "computed: true".

I wonder could we fix these in version 3 ?